ec2.tf

```diff
@@ -103,7 +103,7 @@ module "autoscale_group" {
 
 resource "aws_ecs_capacity_provider" "ec2" {
   for_each = local.ec2_capacity_providers
-  name     = each.key
+  name     = each.value["name"] != null && each.value["name"] != "" ? each.value["name"] : each.key
```

Comment:
```suggestion
  name     = coalesce(each.value["name"], each.key)
```
We can simplify this by using the `coalesce` function, which returns the first non-null, non-empty string from its arguments

Reply:
@gberenice thanks for the tip